src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,38 +3,32 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
-    private Object[] keys = new Object[10];
-    private Object[] values = new Object[10];
+    private K[] keys = (K[]) new Object[10];
+    private V[] values = (V[]) new Object[10];
+    private int index;
 
     @Override
     public void put(K key, V value) {
-        if (get(key) == null) {
-            for (int i = 0; i < keys.length; i++) {
-                if (keys[i] == null & values[i] == null) {
-                    keys[i] = key;
-                    values[i] = value;
-                    break;
-                }
-            }
-        } else {
-            for (int i = 0; i < keys.length; i++) {
-                if (key.equals(keys[i])) {
-                    keys[i] = key;
-                    values[i] = value;
-                    break;
+        for (int i = 0; i < keys.length; i++) {
+            if (key == keys[i] || key != null && key.equals(keys[i])) {
+                keys[i] = key;
+                values[i] = value;
+                if (key == null) {
+                    index++;
                 }
+                return;
             }
         }
+        keys[index] = key;
+        values[index] = value;
+        index++;
     }
 
     @Override
     public V get(K key) {
         V value = null;
         for (int i = 0; i < keys.length; i++) {
-            if (key != null && key.equals(keys[i])) {
-                value = (V) values[i];
-                break;
-            } else if (key == keys[i]) {
+            if (key == keys[i] || key != null && key.equals(keys[i])) {
                 value = (V) values[i];
```

Comment:
```suggestion
                return (V) values[i];
```
і вкінці додати `return null`